### PR TITLE
Add reset service

### LINKED ISSUE #26: Add reset button to reset all resources and scheduler configuration
Add reset button

- delete all resources
- change scheduler configuration to default one.

 need to change both frontend and backend.

/kind feature

### PR DESCRIPTION
**Pull request by sivchari**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
I added reset service to reset resources.

#### Which issue(s) this PR fixes:

Fixes #26

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: server/handler/schedulerconfig.go [hunk 1]
```diff
@@ -39,12 +39,3 @@ func (h *SchedulerConfigHandler) ApplySchedulerConfig(c echo.Context) error {
 
 	return c.NoContent(http.StatusAccepted)
 }
-
-func (h *SchedulerConfigHandler) ResetScheduler(c echo.Context) error {
```
Comment (sanposhiho) [inline server/handler/schedulerconfig.go:43]: R.I.P

File: reset/reset.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type StorageClassService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type PriorityClassService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
```
Comment (sanposhiho) [inline reset/reset.go:28]: They all have the same interface. So, we can use just one interface for better abstraction. 
That way, when new resource is supported in the future, the changes to the Reset service will be small.

File: reset/reset.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		pvcService:   pvcService,
+		scSerivce:    scService,
+		pcService:    pcService,
+		schedService: schedService,
+	}
+}
+
+// Reset cleans up all resources and scheduler configuration.
+func (s *Service) Reset(ctx context.Context) error {
+	lopts := metav1.ListOptions{
```
Comment (sanposhiho) [inline reset/reset.go:68]: Looks not needed. Just passing emptyListOpts to all services should be enough.

**Review by sanposhiho — COMMENTED** [review]
Thanks!

**Comment by sanposhiho** [comment]
@sivchari 
Could you add the brief doc to this api.md?
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docs/api.md

/kind api-change

File: reset/reset.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+// Service cleans up resources.
+type Service struct {
+	client       clientset.Interface
+	nodeService  NodeService
+	pvService    PersistentVolumeService
+	pvcService   PersistentVolumeClaimService
+	scSerivce    StorageClassService
+	pcService    PriorityClassService
+	schedService SchedulerService
+}
```
Comment (sanposhiho) [inline reset/reset.go:43]: With [this](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/105/files#r805167412) change, the field will be like this:

```suggestion
type Service struct {
	client       clientset.Interface
	// deleteServices has the all services for each resource.
	// key: service name. 
	deleteServices map[string]DeleteService
	schedService SchedulerService
}
```

**Comment by sivchari** [comment]
Thank !

File: server/di/di.go [hunk 5]
```diff
@@ -46,6 +48,16 @@ func NewDIContainer(client clientset.Interface, restclientCfg *restclient.Config
...
 	c.priorityClassService = priorityclass.NewPriorityClassService(client)
 	c.exportService = export.NewExportService(client, c.podService, c.nodeService, c.pvService, c.pvcService, c.storageClassService, c.priorityClassService, c.schedulerService)
+
+	deleteServices := map[string]reset.DeleteService{
+		"node": c.nodeService,
+		"pod":  c.podService,
+		"pv":   c.pvService,
+		"pvc":  c.pvcService,
+		"sc":   c.storageClassService,
+		"pc":   c.priorityClassService,
```
Comment (sanposhiho) [inline server/di/di.go:58]: Could you change it to use the full names of resources for better loggings? (like "Persistent volume")

**Review by sanposhiho — COMMENTED** [review]
Sorry, one more nits 🙏  Looks good otherwise.

File: docs/api.md [hunk 6]
```diff
@@ -38,13 +38,13 @@ empty
 | 202   | |
 | 500 | something went wrong (see logs of the simulator server) |
 
-## Reset scheduler configuration
+## Reset all resources and scheduler configutarion
 
-restart scheduler with default configuration.
+clean up all resources and scheduler configuration.
```
Comment (sanposhiho) [inline docs/api.md:43]:
```suggestion
clean up all resources and restore the initial scheduler configuration.
(If you didn't pass the initial scheduler configuration via `KUBE_SCHEDULER_CONFIG_PATH`, the default scheduler configuration will be restored.)
```

**Review by sanposhiho — COMMENTED** [review]
sorry, "one more nits" again.. 🤦‍♂️

File: docs/api.md (hunk 6 above)
Comment (sivchari) [reply docs/api.md]: Ok, I fixed it.

**Review by sanposhiho — APPROVED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**